### ISSUE TITLE
Kover breaks mocking OkHttp's Response with MockK

### ISSUE THREAD

**Issue by cristan** [Kotlin/kotlinx-kover#88]
Tested on Kover 0.4.4:
```
plugins {
    id 'org.jetbrains.kotlinx.kover' version '0.4.4'
}
```

Dependencies:

```
implementation 'com.squareup.okhttp3:okhttp:4.9.1'
testImplementation "io.mockk:mockk:1.12.1"
```

Tests:
```
@Test
fun testMockingRequest() {
    val response: Request = mockk()
}

@Test
fun testMockingResponse() {
    val response: Response = mockk()
}
```

`testMockingResponse()` fails here with this stacktrace:
```
java.lang.VerifyError
	at java.instrument/sun.instrument.InstrumentationImpl.retransformClasses0(Native Method)
	at java.instrument/sun.instrument.InstrumentationImpl.retransformClasses(InstrumentationImpl.java:167)
	at io.mockk.proxy.jvm.transformation.JvmInlineInstrumentation.retransform(JvmInlineInstrumentation.kt:28)
	at io.mockk.proxy.common.transformation.RetransformInlineInstrumnetation$execute$1.invoke(RetransformInlineInstrumnetation.kt:19)
	at io.mockk.proxy.common.transformation.RetransformInlineInstrumnetation$execute$1.invoke(RetransformInlineInstrumnetation.kt:6)
	at io.mockk.proxy.common.transformation.ClassTransformationSpecMap.applyTransformation(ClassTransformationSpecMap.kt:41)
	at io.mockk.proxy.common.transformation.RetransformInlineInstrumnetation.execute(RetransformInlineInstrumnetation.kt:16)
	at io.mockk.proxy.jvm.ProxyMaker.inline(ProxyMaker.kt:88)
	at io.mockk.proxy.jvm.ProxyMaker.proxy(ProxyMaker.kt:30)
	at io.mockk.impl.instantiation.JvmMockFactory.newProxy(JvmMockFactory.kt:34)
	at io.mockk.impl.instantiation.AbstractMockFactory.newProxy$default(AbstractMockFactory.kt:29)
	at io.mockk.impl.instantiation.AbstractMockFactory.mockk(AbstractMockFactory.kt:59)
	at com.example.testkoverandroidproject.ExampleUnitTest.testMockingResponse(ExampleUnitTest.kt:73)
```

This doesn't happen when I don't add Kover as a plugin. Surprisingly enough, `testMockingRequest()` always succeeds, so there's something in in mocking `Response` which goes wrong which doesn't happen when mocking `Request`.

**Comment by shanshin** [comment]
Hi,
to eliminate the conflict between the MockK and the Kover, try to add all OkHttp classes to the instrumentation exclusions
```
    testOptions {
        unitTests.all {
            kover {
                excludes = ['okhttp3\\..+']
            }
        }
    }
```

**Comment by cristan** [comment]
It's a little weird that this is needed, but excluding OkHttp classes indeed works.

**Comment by cristan** [comment]
Another instance where mocking okhttp stuff is broken by kover:

```
@Test
fun `test mocking proceeding a chain`() {
    val testChain = mockk<Interceptor.Chain>()
    val argument = slot<Request>()
    every { testChain.proceed(capture(argument)) }.returns(mockk())
}
```

This gives this error:
```
java.lang.VerifyError
	at java.instrument/sun.instrument.InstrumentationImpl.retransformClasses0(Native Method)
	at java.instrument/sun.instrument.InstrumentationImpl.retransformClasses(InstrumentationImpl.java:167)
	at io.mockk.proxy.jvm.transformation.JvmInlineInstrumentation.retransform(JvmInlineInstrumentation.kt:28)
```

Even excluding all okhttp classes doesn't resolve this issue.

**Comment by shanshin** [comment]
@foo4u, in version `0.5.0`, the format of filters for classes has changed. Now, instead of specifying a regular expression, you can use only wildcards `*` and `?`. 
Therefore, now you can exclude OkHttp from instrumentation like this:
```
    testOptions {
        unitTests.all {
            kover {
                excludes = ['okhttp3.*']
            }
        }
    }
```

**Comment by louiseightsix** [comment]
Thanks for providing a workaround; this is affecting us as well - would love to get to the bottom of it.

**Comment by shanshin** [comment]
There are two ways to avoid this problem:
- exclude specific libraries from the instrumentation by hardcode their packages - in this case, we do not know in advance which libraries cause problems during instrumentation, and errors will be repeated.
- exclude from instrumentation all classes that are not included in the user project

**Comment by shanshin** [comment]
Closed in favor of #418
